components/header/navbar/searchpage/SearchBar.tsx

```diff
@@ -0,0 +1,67 @@
+import ActivityIndicator from '@components/ActivityIndicator'
+import React, { useEffect } from 'react'
+import { MdCancel, MdSearch } from 'react-icons/md'
+
+type SearchBarProps = {
+  query: string
+  handleSearch: () => void
+  setQuery: (query: string) => void
+  clearResults: () => void
+  searching: boolean
+  setIsFetched: (isFetched: boolean) => void
+}
+
+const SearchBar = ({
+  query,
+  handleSearch,
+  setQuery,
+  clearResults,
+  searching,
+  setIsFetched,
+}: SearchBarProps) => {
+  useEffect(() => {
+    setIsFetched(false)
+    const delayDebounceFn = setTimeout(() => {
+      if (query) {
+        handleSearch()
+      }
+    }, 500) // 500ms delay
+    return () => clearTimeout(delayDebounceFn)
+  }, [query])
```

Comment:
This `useEffect` is not needed. You should be able to do the debounce in the event handler.